components/layout_thread/lib.rs

```diff
@@ -1019,18 +1020,20 @@ impl LayoutThread {
             }
             let display_list = (*rw_data.display_list.as_ref().unwrap()).clone();
 
+            debug!("Layout done!");
+
+            // TODO: Avoid the temporary conversion and build webrender sc/dl directly!
+            let wr_dl = rw_data.display_list.as_ref().unwrap().convert_to_webrender(self.id).finalize();
+
             if opts::get().dump_display_list {
-                display_list.print();
+                println!("// Display List from {:?}\n{}",
```

Comment:
Looking at this now, I'm wondering if it's better to make WebRender responsible for printing the display list? Or is WebRender already doing a lot of the work here?

With this change, can't we remove the printing methods from the DisplayList struct as well?

Reply:
> Looking at this now, I'm wondering if it's better to make WebRender responsible for printing the display list? Or is WebRender already doing a lot of the work here?

I am not sure if I understand these questions. WebRender has the types and visitors for serialization. The RON crate formats it to actual text and servo just prints it. The comment heading `Display List from {:?}`is just because I found it somewhat difficult to see where one display list ends and another starts.

> With this change, can't we remove the printing methods from the DisplayList struct as well?
Yes we can. This was the WIP part of the PR.